src/gt4py/backend/gtc_backend/defir_to_gtir.py

```diff
@@ -185,6 +196,14 @@ def visit_TernaryOpExpr(self, node: TernaryOpExpr) -> gtir.TernaryOp:
             false_expr=self.visit(node.else_expr),
         )
 
+    def visit_BuiltinLiteral(self, node: BuiltinLiteral) -> gtir.Literal:  # type: ignore[return]
+        # currently deals only with boolean literals
+        if node.value in self.GT4PY_BUILTIN_TO_GTIR.keys():
+            return gtir.Literal(
+                value=self.GT4PY_BUILTIN_TO_GTIR[node.value], dtype=common.DataType.BOOL
+            )
+        assert f"BuiltIn.{node.value} not implemented in lowering"
```

Comment:
This line seems to have no effect under any circumstances? The string always evaluates to true.

Reply:
ups.

Reply:
replaced by `raise NotImplementedError(f"BuiltIn.{node.value} not implemented in lowering")`. Good?